From e8bc9984bbf6d2a999ab7aaa3392929659d9d916 Mon Sep 17 00:00:00 2001
From: ggarand <ggarand@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Fri, 12 Feb 2010 15:41:35 +0000
Subject: [PATCH] automatically merged revision 1089159:
 Critical performance fix:

There is a huge difference of behaviour between DEBUG and RELEASE
builds of Qt 4.6, that breaks scrolling in KHTML.

Using QWidget::scroll() on a widget that doesn't have WA_OpaquePaintEvent
attribute set works just fine in Debug, but completely breaks down in
Release, causing the whole view to get repainted always.


CCMAIL: kde-packager@kde.org

git-svn-id: svn+ssh://svn.kde.org/home/kde/branches/KDE/4.4/kdelibs@1089160 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 khtml/khtmlview.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/khtml/khtmlview.cpp b/khtml/khtmlview.cpp
index c2481e2..4a22190 100644
--- a/khtml/khtmlview.cpp
+++ b/khtml/khtmlview.cpp
@@ -604,6 +604,12 @@ void KHTMLView::init()
         setWidget( new QWidget(this) );
     widget()->setAttribute( Qt::WA_NoSystemBackground );
 
+    // Do *not* remove this attribute frivolously.
+    // You might not notice a change of behaviour in Debug builds
+    // but removing opaque events will make QWidget::scroll fail horribly
+    // in Release builds.
+    widget()->setAttribute( Qt::WA_OpaquePaintEvent );
+
     verticalScrollBar()->setCursor( Qt::ArrowCursor );
     horizontalScrollBar()->setCursor( Qt::ArrowCursor );
 
-- 
1.7.0

